calcom/cal.com
# fix: wrong slot starting times for half hour times zones

## What does this PR do?

Fixes that UTC was used to find the time of the first slot. This resulted in wrong slot times for timezones that are not full hours like Asia/Kolkata +5:30.

#### Example: 

availability: 7:30 - 12:00 (+5:30)

Slots for hourly event before: 
<img width="281" alt="Screenshot 2023-07-14 at 09 53 10" src="https://github.com/calcom/cal.com/assets/30310907/b42c6908-26ef-4f5c-97d3-73bac5f3a5b2">

Slots for hourly event after: 
<img width="278" alt="Screenshot 2023-07-14 at 09 52 50" src="https://github.com/calcom/cal.com/assets/30310907/4dfc3196-831c-4396-9b8e-f740e06a3a13">

Fixes #10136

## Type of change


## How should this be tested?

 - Check if tests pass 
 - See if slots start at full hours for and hourly event, availability starting at 7:30 in timezone 5:30
